### PR TITLE
[#1762] Icons set ability for `VaButtonToggle`

### LINKED ISSUE #1762: Add slot to `VaButtonToggle`
- [ ] Add slot to `VaButtonToggle`

**Description**:
We want this to let users customize component content.

**Example**:
This way we can have `Icon` instead of the `text` inside of the button, etc.

**More info**:
[Initial community request from discord](https://discord.com/channels/457593300799717377/771405067474567178/973605631888085092).

### PR DESCRIPTION
**Pull request by aluarius**
close #1762 

## Description
- [x] Ability to set icons instead (or in addition) of labels for `VaButtonToggle` has been added.
- [x] Fixed icons margin in `VaButton` in empty default slot was passed.
- [x] Docs & examples updated (fixed some bugs in examples etc).

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-button-toggle/VaButtonToggle.vue [hunk 1]
```diff
@@ -39,45 +39,59 @@ export default defineComponent({
...
-    textColor: { type: String as PropType<string>, default: undefined },
-    activeButtonTextColor: { type: String as PropType<string>, default: 'white' },
+    options: { type: Array, default: () => [] },
+    color: { type: String, default: 'primary' },
+    textColor: { type: String, default: undefined },
+    activeButtonTextColor: { type: String, default: 'white' },
     modelValue: { type: [String, Number] as PropType<string | number>, default: '' },
-    outline: { type: Boolean as PropType<boolean>, default: false },
-    flat: { type: Boolean as PropType<boolean>, default: false },
-    rounded: { type: Boolean as PropType<boolean>, default: true },
```
Comment (m0ksem) [inline packages/ui/src/components/va-button-toggle/VaButtonToggle.vue:49]: Please, remove PropTypes in different PR next time to have a smaller PR diff :)

File: packages/ui/src/components/va-button-toggle/VaButtonToggle.vue [hunk 2]
```diff
@@ -39,45 +39,59 @@ export default defineComponent({
...
-      if (buttonValue !== props.modelValue) { return }
+    const isFlatOrOutline = computed(() => props.outline || props.flat)
+    const color = computed(() => {
+      if (props.toggleColor) {
+        return toggleColorComputed.value
+      } else {
+        return isFlatOrOutline.value ? colorComputed.value : shiftHSLAColor(colorComputed.value, { l: -6 })
+      }
+    })
+    const textColor = computed(() => props.activeButtonTextColor || getTextColor(colorComputed.value))
```
Comment (m0ksem) [inline packages/ui/src/components/va-button-toggle/VaButtonToggle.vue:73]: Look like issue with text colors fixed in this PR: https://github.com/epicmaxco/vuestic-ui/pull/1769.

Again, let's have a smaller diff.

File: packages/ui/src/components/va-button-toggle/VaButtonToggle.demo.vue [hunk 3]
```diff
@@ -249,6 +266,16 @@ export default {
         { label: 'Two', value: 'two' },
         { label: 'Three', value: 'three' },
       ],
+      optionsWithIcons: [
+        { label: 'One', value: 'one', icon: 'home' },
+        { label: 'Two', value: 'two', iconRight: 'home' },
+        { label: 'Three', value: 'three', icon: 'home', iconRight: 'home' },
```
Comment (m0ksem) [inline packages/ui/src/components/va-button-toggle/VaButtonToggle.demo.vue:272]: I personally prefer if this issue will have slot based fix. But it looks nice anyway.

Do we also want to have?: 
```vue
<VaButton label="Click me" />
```

Then options will look pretty as VaButton props. So we can pass any prop in options:
```
options: { type: Array as PropType<(VaButtonProps & { value: any })[]> }
```

This way we make a beautiful IDE support.

**Review by m0ksem — APPROVED** [review]
Looks good. What do you think about my suggestions? 
Please, make sure you don't have fix collisions with useTextColor hook.

**Review by rustem-nasyrov — APPROVED** [review]

**Pull request merged**